src/Neo/Wallets/Helper.cs

```diff
@@ -126,35 +128,73 @@ public static long CalculateNetworkFee(this Transaction tx, DataCache snapshot,
+                                    script.EmitPush(new byte[64]);
+                                    break;
+                                case ContractParameterType.Boolean:
+                                    script.EmitPush(true);
+                                    break;
+                                case ContractParameterType.Integer:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    break;
+                                case ContractParameterType.Hash160:
+                                    script.EmitPush(new byte[UInt160.Length]);
+                                    break;
+                                case ContractParameterType.Hash256:
+                                    script.EmitPush(new byte[UInt256.Length]);
+                                    break;
+                                case ContractParameterType.PublicKey:
+                                    script.EmitPush(new byte[33]);
+                                    break;
+                                case ContractParameterType.Array:
+                                    script.Emit(OpCode.PUSHINT256, new byte[Integer.MaxSize]);
+                                    script.Emit(OpCode.NEWARRAY);
+                                    break;
+                                default:
+                                    throw new ArgumentException("The verify method requires parameters that need to be passed via the witness' invocation script.");
+                            }
+                        }
+                        invocationScript = script.ToArray();
+                    }
 
                     // Empty verification and non-empty invocation scripts
-                    var invSize = invocationScript?.GetVarSize() ?? Array.Empty<byte>().GetVarSize();
-                    size += Array.Empty<byte>().GetVarSize() + invSize;
+                    var invSize = invocationScript?.GetVarSize() ?? System.Array.Empty<byte>().GetVarSize();
+                    size += System.Array.Empty<byte>().GetVarSize() + invSize;
 
                     // Check verify cost
                     using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, tx, snapshot.CreateSnapshot(), settings: settings, gas: maxExecutionCost);
                     engine.LoadContract(contract, md, CallFlags.ReadOnly);
                     if (invocationScript != null) engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
-                    if (engine.Execute() == VMState.FAULT) throw new ArgumentException($"Smart contract {contract.Hash} verification fault.");
-                    if (!engine.ResultStack.Pop().GetBoolean()) throw new ArgumentException($"Smart contract {contract.Hash} returns false.");
+                    _ = engine.Execute(); // https://github.com/neo-project/neo/issues/2805
```

Comment:
```suggestion
                    if (engine.Execute() == VMState.FAULT) throw new ArgumentException($"Smart contract {contract.Hash} verification fault.");
_ = engine.ResultStack.Pop().GetBoolean(); // The result must be convertible to boolean
```

Reply:
Yes, that's right, but here we should add one more check that there's no extra items on stack.

Reply:
Exactly like a standard verification code.

Reply:
Done

Reply:
If we can reuse the VerifyWitness from [neo/src/Neo/SmartContract/Helper.cs](https://github.com/neo-project/neo/blob/3351533e806d83cad68c93cd3593f34c16223b1b/src/Neo/SmartContract/Helper.cs#L361-L363), it would be much better.